File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        //use binary search to do this
        
        int maxOutput = 0;
        
        //1. Initialize the pointers
        int leftPointer = 0;
        int rightPointer = height.length-1;
        
        //2. While loop 
        while(leftPointer<=rightPointer){
            //3. Intialize the max area stuff
            int maxArea = Math.min(height[leftPointer], height[rightPointer]) * (rightPointer - leftPointer);
            maxOutput = Math.max(maxArea, maxOutput);
            
            if(height[leftPointer] < height[rightPointer]){
                leftPointer++;
            }else{
                rightPointer--;
            }
        }
        
        return maxOutput;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        maxValue = 0
        
        left = 0
        right = len(height)-1
        
        
        while left<=right:
            currentArea = min(height[left], height[right]) *(right-left)
            maxValue = max(currentArea, maxValue)
            
            if height[left]<height[right]:
                left+=1
            else:
                right-=1
        
        
        return maxValue
        
        